base/sparse/sparsematrix.jl

```diff
@@ -1390,7 +1390,77 @@ function one{T}(S::SparseMatrixCSC{T})
     speye(T, m)
 end
 
-## Unary arithmetic and boolean operators
+
+## Broadcast operations involving a single sparse matrix and possibly broadcast scalars
+
+function broadcast{Tf}(f::Tf, A::SparseMatrixCSC)
+    fofzero = f(zero(eltype(A)))
+    fpreszero = fofzero == zero(fofzero)
+    return fpreszero ? _broadcast_zeropres(f, A) : _broadcast_notzeropres(f, A)
+end
+"""
+Returns a `SparseMatrixCSC` that shares `A`'s structure but with stored entries
+`broadcast(f, A.nzval)`.
+"""
+function _broadcast_zeropres{Tf}(f::Tf, A::SparseMatrixCSC)
+    Bcolptr = copy(A.colptr); resize!(Bcolptr, A.n + 1)
+    Browval = copy(A.rowval); resize!(Browval, nnz(A))
+    Bnzval = broadcast(f, A.nzval); resize!(Bnzval, nnz(A))
+    return SparseMatrixCSC(A.m, A.n, Bcolptr, Browval, Bnzval)
+    # TODO: The following lines for constructing Bcolptr and Browval might be safer than
+    # those above: If A.colptr and/or A.rowval are shorter than their expected lengths, the
+    # copy!s might catch the error, whereas the former copy/resize! combinations won't. The
+    # lines above are simpler/cleaner though. Thoughts?
+    # Bcolptr = similar(A.colptr, A.n + 1)
+    # copy!(Bcolptr, 1, A.colptr, 1, A.n + 1)
+    # Browval = similar(A.rowval, nnz(A))
+    # copy!(Browval, 1, A.rowval, 1, nnz(A))
```

Comment:
These alternative lines for constructing `Bcolptr` and `Browval` might be safer than the `copy`/ `resize!` lines above: If `A.colptr` and/or `A.rowval` are shorter than their expected lengths, the `copy!`s might catch the error, whereas the former `copy`/`resize!` combinations won't. The `copy`/`resize!` lines are simpler/cleaner though. Thoughts? Thanks!

These alternative lines for constructing `Bcolptr` and `Browval` might be safer